packages/babel-preset-env/src/polyfills/corejs2/usage-plugin.js

```diff
@@ -0,0 +1,224 @@
+import corejs2Polyfills from "../../../data/corejs2-built-ins.json";
+import getPlatformSpecificDefaultFor from "./get-platform-specific-default";
+import filterItems from "../../filter-items";
+import {
+  BuiltIns,
+  StaticProperties,
+  InstanceProperties,
+} from "./built-in-definitions";
+import {
+  createImport,
+  getType,
+  has,
+  isPolyfillSource,
+  isPolyfillRequire,
+} from "../../utils";
+import { logUsagePolyfills } from "../../debug";
+
+const NO_DIRECT_POLYFILL_IMPORT = `
+  When setting \`useBuiltIns: 'usage'\`, polyfills are automatically imported when needed.
+  Please remove the \`import '@babel/polyfill'\` call or use \`useBuiltIns: 'entry'\` instead.`;
+
+export default function(
+  { types: t },
+  { include, exclude, polyfillTargets, debug },
+) {
+  const polyfills = filterItems(
+    corejs2Polyfills,
+    include,
+    exclude,
+    polyfillTargets,
+    getPlatformSpecificDefaultFor(polyfillTargets),
+  );
+
+  const addAndRemovePolyfillImports = {
+    ImportDeclaration(path) {
+      if (
+        path.node.specifiers.length === 0 &&
+        isPolyfillSource(path.node.source.value)
+      ) {
+        console.warn(NO_DIRECT_POLYFILL_IMPORT);
```

Comment:
I think that this would warn providing the context, but it needs to be checked.

```js
console.warn(
  path.buildCodeFrameError(NO_DIRECT_POLYFILL_IMPORT)
)
```